app/src/main/java/com/github/geohunt/app/authentication/Authenticator.kt

```diff
@@ -0,0 +1,27 @@
+package com.github.geohunt.app.authentication
+
+import androidx.activity.ComponentActivity
+import com.github.geohunt.app.model.database.api.User
+import java.util.concurrent.CompletableFuture
+
+interface Authenticator {
+
+    val user: User?
+
+    /**
+     * Begins an authentication phase from the given activity.
+     * Returns a future that will be completed when the user will be logged in.
+     * If the authentication fails, the completable future will fail.
+     * @param activity The activity that requests the connection
+     * @return A completable future that follows the above description
+     */
+    fun authenticate(activity: ComponentActivity): CompletableFuture<User>
+
+    /**
+     * Signs out the user.
+     * @param activity The activity that requests the signing out
+     * @return The completable future is finished when the signing out is finished.
+     */
+    fun signOut(activity: ComponentActivity): CompletableFuture<Void>
+
```

Comment:
Once the database #36 will be merged I feel like this should be linked to the Database interface somehow. This is because the database rely on the fact that it is connected as a certain user. I don't know how to achieve this currently but it is worth keeping in mind

Reply:
I do not agree. Those are two distincts objects. If the database need a user, it should be either taken globally as a parameter (like the service locator or singleton pattern).

Reply:
Yeah, ok you're right